File: C/Algorithms/1752.c
#include <stdbool.h>

bool check(int *nums, int numsSize) {
  int start = nums[0];
  for (int i = 1; i < numsSize; i++) {
    // posible point of rotation
    if (nums[i - 1] > nums[i]) {
      for (int j = i + 1; j < numsSize; j++) {
        // Make sure we still have non-decreasing order
        // and none are larger than start (which would
        // mean that before rotation, nums[i-1] < nums[i] for
        // nums[i-1] == start.)
        if (nums[j - 1] > nums[j] || start < nums[j])
          return false;
      }
    }
  }
  return true;
}

File: Python/Algorithms/1752.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        start = nums[0]
        for i in range(1, len(nums)):
            # iterate until you find possible point of rotation.
            if nums[i-1] > nums[i]:
                # then go through the rest of the elements and
                # check that they are in correct order *and*
                # that they are smaller than start (which is
                # equal to nums[0]).
                for i in range(i+1, len(nums)):
                    if nums[i] > start or nums[i-1] > nums[i]:
                        return False
        return True
